client/landing/stepper/declarative-flow/internals/steps-repository/create-site/index.tsx

```diff
@@ -232,7 +232,8 @@ const CreateSite: StepType = function CreateSite( { navigation, flow, data } ) {
 			siteIntent,
 			undefined, // siteGoals
 			gardenName,
-			gardenPartnerName
+			gardenPartnerName,
+			urlQueryParams.get( 'spec_id' ) || ''
```

Comment:
Probably doesn't really matter, but should the fallback default be `null` rather than empty string?